packages/@o3r/schematics/src/utility/dependencies.ts

```diff
@@ -8,9 +9,14 @@ import { NodeDependency, NodeDependencyType } from '@schematics/angular/utility/
  *  @returns The version range value retrieved from the provided package.json file
  */
 export function getExternalDependenciesVersionRange<T extends string>(packageNames: T[], packageJsonPath: string): Record<T, string> {
-  const packageJsonContent = JSON.parse(fs.readFileSync(packageJsonPath, {encoding: 'utf-8'}));
+  const packageJsonContent = JSON.parse(fs.readFileSync(packageJsonPath, {encoding: 'utf-8'})) as PackageJson & { generatorDependencies: Record<string, string> };
   return packageNames.reduce((acc: Partial<Record<T, string>>, packageName) => {
-    acc[packageName] = packageJsonContent.peerDependencies?.[packageName] || packageJsonContent.generatorDependencies?.[packageName] || packageJsonContent.dependencies?.[packageName] || 'latest';
+    acc[packageName] =
+      packageJsonContent.peerDependencies?.[packageName] ||
+      packageJsonContent.generatorDependencies?.[packageName] ||
```

Comment:
the `generatorDependencies` should be the first in the priority order actually (to handle the case we do not want it to be aligned with devDeps for example)